libweb3jsonrpc/Test.cpp

```diff
@@ -23,16 +23,39 @@
 #include "Test.h"
 #include <jsonrpccpp/common/errors.h>
 #include <jsonrpccpp/common/exception.h>
-#include <libethereum/ClientTest.h>
+#include <libdevcore/CommonJS.h>
 #include <libethereum/ChainParams.h>
+#include <libethereum/ClientTest.h>
 
 using namespace std;
 using namespace dev;
+using namespace dev::eth;
 using namespace dev::rpc;
 using namespace jsonrpc;
 
 Test::Test(eth::Client& _eth): m_eth(_eth) {}
 
+string exportLog(eth::LogEntries const& _logs)
+{
+    RLPStream s;
+    s.appendList(_logs.size());
+    for (eth::LogEntry const& l : _logs)
+        l.streamRLP(s);
+    return toHexPrefixed(sha3(s.out()));
+}
+
+string Test::test_getLogHash(string const& _txHash)
+{
+    if (m_eth.blockChain().isKnownTransaction(h256(_txHash)))
```

Comment:
What happens when you send here invalid hash string? Does it crash or is appropriate error returned?

Reply:
```
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<dev::BadHexCharacter>'
  what():  /home/wins/Ethereum/cpp-ethereum/libdevcore/CommonData.cpp(93): Throw in function dev::bytes dev::fromHex(const string&, dev::WhenError)
Dynamic exception type: boost::exception_detail::clone_impl<dev::BadHexCharacter>
```

Reply:
So you should catch it here and rethrow as `JsonRpcException`, then method will correctly return error.